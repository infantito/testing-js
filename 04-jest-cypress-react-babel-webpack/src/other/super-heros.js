const superHeros = [{powers: 'fly x'}, {powers: 'fly y'}, {powers: 'fly z'}]

function getFlyingSuperHeros() {
  return superHeros.filter(hero => {
    return hero.powers.includes('fly')
  })
}

export {getFlyingSuperHeros}
