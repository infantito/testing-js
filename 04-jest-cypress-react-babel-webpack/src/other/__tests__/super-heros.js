import {getFlyingSuperHeros} from '../super-heros'

test('returns super heros that can fly', () => {
  const flyingHeros = getFlyingSuperHeros()

  expect(flyingHeros).toEqual([
    {
      powers: 'fly x',
    },
    {
      powers: 'fly y',
    },
  ])
})
