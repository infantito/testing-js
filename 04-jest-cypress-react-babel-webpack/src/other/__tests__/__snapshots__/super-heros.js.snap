// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`returns super heros that can fly 1`] = `
Array [
  Object {
    "powers": "fly x",
  },
  Object {
    "powers": "fly y",
  },
  Object {
    "powers": "fly z",
  },
]
`;
